Improve getting started (#325)
mautic/documentation: en/setup/getting_started.md

## Step 1: Install Mautic

If you have already downloaded the zip from the download page or have installed Mautic through some other source (Softaculous, Bitnami, Digital Ocean etc...) then you have already completed the first step. If not then you will need to upload the Mautic package (a zip file) to your server; unzip the files; and then navigate to that location in your browser. 

As an example, if your domain is mydomain.com and you have unzipped the file into a folder called 'm', you would then browse to mydomain.com/m to start the installation process.

###Permission errors
If you see errors when you try to access the installation as below:

![](/setup/media/permissions-error.png)

the problem is likely to be due to the permissions on your files and folders.  

If you have access to the command line, you can run the following commands to reset the file and folder permissions.  Make sure that you are in the directory where Mautic has been installed (using the example above, that would mean executing these commands within /m in your web root folder - you can get there using the `cd` command)

> ```find . -type d -exec chmod 755 {} \;

Reviewer: I believe the 3 starting and ending backticks must be on a single line so the Markdown parser would style the content in between as a code sample.

What's wrong with this PR, Ruth? What shouldn't be part of this PR?

Author: ![screenshot-github com-2019 09 02-10_07_41](https://user-images.githubusercontent.com/2930593/64103251-b5fc9400-cd69-11e9-80f5-9b3b76f508f9.png) I see this on the PR which I don't think should be there (it was me bringing my fork up to date with the upstream repo).

In regards the backticks I was following https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code so I guess I need to add a line break after the final line?

Reviewer: Yes, there should be a line break.

I don't think the merge commit harms anything. I refresh the master branch to ensure it is exactly the same as the remote one like this:
`$ git fetch upstream`
`$ git reset --hard upstream/master`
⚠️ but it will delete all changes and commits that are not in the remote master branch.

Reviewer 2: You are right that the merge commit shouldn't harm anything.  However it is nicer to avoid it if possible since it does make the history more complicated to look at.  I provided a suggestion below for using rebase to get rid of the merge.

find . -type f -exec chmod 644 {} \;
chmod -R g+w app/cache/
chmod -R g+w app/logs/
chmod -R g+w app/config/
chmod -R g+w media/files/
chmod -R g+w media/images/
chmod -R g+w translations/

> chown -R $USER:www-data .```

Reviewer 2: For this kind of code block, I would suggest avoiding use of triple backticks and instead simply indenting the whole thing by 4 columns:

    This is some normal text.

        find . -type f -exec chmod 644 {} \;
        chmod -R g+w app/cache/

    and normal text again.

This will achieve exactly the same rendering effect, but has the advantage of making the non-rendered plain-text file much more readable.  This adheres to Markdown's original design goal, as stated on [the homepage](https://daringfireball.net/projects/markdown/):

> The overriding design goal for Markdown’s formatting syntax is to make it as readable as possible.

However, if you do use triple backticks, bear in mind the following:

- [Any text immediately following the opening backticks on the first line is used by GitHub to indicate the coding language of the rest of the block, so that it can syntax highlight correctly](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown)
- For single-line code blocks, single backticks are sufficient.  Triple backticks are only needed when the content contains backticks of its own.

As an example, if your domain is mydomain.com and you have unzipped the file into a folder called 'm', you would then browse to mydomain.com/m to start the installation process.

### Permission errors
If you see errors when you try to access the installation as below:

![](/setup/media/permissions-error.png)

the problem is likely to be due to the permissions on your files and folders.  

If you have access to the command line, you can run the following commands to reset the file and folder permissions.  Make sure that you are in the directory where Mautic has been installed (using the example above, that would mean executing these commands within /m in your web root folder - you can get there using the `cd` command)

```find . -type d -exec chmod 755 {} \;
find . -type f -exec chmod 644 {} \;
chmod -R g+w app/cache/
chmod -R g+w app/logs/
chmod -R g+w app/config/
chmod -R g+w media/files/
chmod -R g+w media/images/
chmod -R g+w translations/

> chown -R $USER:www-data .

Reviewer 3: ```suggestion
```

This step may take a few minutes to complete, as the database is created and populated.

### Admin user
The next step after creating the database is to set up your administrator account.  This will be the account you use to log into Mautic for the first time.

![](/setup/media/admin-user.png)

Enter a username, password, first and last name and an email address, and the account will be created for you.

### Email configuration

The next step allows you to configure the way in which email is sent from your Mautic instance.  

![](/setup/media/email-config.png)

If you are testing, whether locally or on a live server, it is highly recommended to make use of [Mailhog](https://github.com/mailhog/MailHog) or similar to capture all outgoing emails, until you are ready to send to real recipients. Otherwise, in a production environment you will need to choose whether you want to send email immediately, or queue email (and process it when the Cron job runs), and provide the appropriate server, port, encryption type, credentials and authentication mode for the systems you wish to use.

> ###Admin login

Reviewer 3: ```suggestion
### Admin login

```

> Now you have completed the setup process, you will be directed to the back-end login screen at mydomain.com/s/login - this is where you log into Mautic to manage your instance, so it's worth bookmarking the login page for future reference.

Reviewer 3: ```suggestion
Now you have completed the setup process, you will be directed to the back-end login screen at `http://mydomain.com/s/login` - this is where you log into your Mautic to manage your instance, so it's worth bookmarking the login page for future reference.
```

## Step 2: Add Cron Jobs

> Once you've installed Mautic you will need to create a few standard cron jobs to have your software process various tasks. These cron jobs can be created through a cPanel or added through command line. If you are unfamiliar or uncomfortable with this step then we'd recommend asking in the forums or in the live Slack chat. Here is a list of the cron jobs you'll need to create. (Please note: "/path/to/mautic..." will reflect the directory where your version of Mautic is installed, and you may need to replace 'php' with the path to PHP on your server. Ask your hosting provider if you are not sure.)

Reviewer 3: ```suggestion
Once you've installed Mautic you will need to create a few standard [cron jobs][cron] to have your software process various tasks. These [cron jobs][cron] can be created through a cPanel or added through command line. If you are unfamiliar or uncomfortable with this step then we'd recommend asking in the forums or in the live Slack chat. Here is a list of the [cron jobs][cron] you'll need to create. (Please note: "/path/to/mautic..." will reflect the directory where your version of Mautic is installed, and you may need to replace 'php' with the path to PHP on your server. Ask your hosting provider if you are not sure.)

[cron][cron_jobs.md]

```

### Question

Is `[cron][cron_jobs.md]` or `[cron][cron_jobs.html]` the correct link here?  ie does `gitbook` translate a `.md` file link to the post-rendered `.html`

Reviewer 3: ### Answer

    [cron][cron_jobs.html]

![](/setup/media/config-menu.png)

On the System Settings tab, scroll down to find the IP lookup service option and click the "Fetch IP Lookup Data Store."

![](/setup/media/get-maxmind.png)

You could also choose another supported IP lookup service if you prefer.

## Step 4: Install the Tracking Javascript

After installation and setup of the cron jobs you're ready to begin tracking contacts. You will need to add a simple javascript to the websites for each site you wish to track via Mautic. This is a very simple process and you can add this tracking script to your website template file, or install a Mautic integration for the more common CMS platforms. Here is an example of the tracking javascript which you can access by clicking on 'Tracking Settings':

``` <script>
    (function(w,d,t,u,n,a,m){w['MauticTrackingObject']=n;
        w[n]=w[n]||function(){(w[n].q=w[n].q||[]).push(arguments)},a=d.createElement(t),
        m=d.getElementsByTagName(t)[0];a.async=1;a.src=u;m.parentNode.insertBefore(a,m)

>     })(window,document,'script','http://mydomain.com/m/mtc.js','mt');

Reviewer 3: ```suggestion
    })(window,document,'script','http://example.com/mautic/mtc.js','mt');
```

    mt('send', 'pageview');
</script>
```

> You will need to change the site URL (replace mydomain.com/m with the URL to your Mautic instance) in the above script.

Reviewer 3: ```suggestion
You will need to change the site URL (replace example.com/mautic with the URL to your Mautic instance) in the above script.
```
